test/RestrictedNORI.test.ts

```diff
@@ -0,0 +1,681 @@
+import { BigNumber } from 'ethers';
```

Comment:
a ton of this code could use a little more DRY (many tests start with the same 10 lines of code), but doesnt need to happen in this PR

Reply:
probably worth thinking comprehensively all at once about how to organize the abstractions now that most of the market functionality we need to test is here.  it's been interesting to watch the test utilities develop